src/item/enchantment/ItemFlags.php

```diff
@@ -29,27 +29,31 @@ private function __construct(){
 		//NOOP
 	}
 
-	//TODO: this should probably move to protocol
-
 	public const NONE = 0x0;
-	public const ALL = 0xffff;
-	public const ARMOR = self::HEAD | self::TORSO | self::LEGS | self::FEET;
-	public const HEAD = 0x1;
-	public const TORSO = 0x2;
-	public const LEGS = 0x4;
-	public const FEET = 0x8;
+	public const ALL = 0xffffff;
+	public const BREAKABLE = self::ALL & ~self::COMPASS & ~self::RECOVERY_COMPASS & ~self::HEAD;
+	public const ARMOR = self::HELMET | self::CHESTPLATE | self::LEGGINGS | self::BOOTS;
+	public const HELMET = 0x1;
+	public const CHESTPLATE = 0x2;
+	public const LEGGINGS = 0x4;
+	public const BOOTS = 0x8;
 	public const SWORD = 0x10;
-	public const BOW = 0x20;
-	public const TOOL = self::HOE | self::SHEARS | self::FLINT_AND_STEEL;
-	public const HOE = 0x40;
-	public const SHEARS = 0x80;
-	public const FLINT_AND_STEEL = 0x100;
-	public const DIG = self::AXE | self::PICKAXE | self::SHOVEL;
-	public const AXE = 0x200;
-	public const PICKAXE = 0x400;
-	public const SHOVEL = 0x800;
-	public const FISHING_ROD = 0x1000;
-	public const CARROT_STICK = 0x2000;
-	public const ELYTRA = 0x4000;
-	public const TRIDENT = 0x8000;
+	public const TRIDENT = 0x20;
+	public const BOW = 0x40;
+	public const CROSSBOW = 0x80;
+	public const SHIELD = 0x100;
+	public const SHEARS = 0x200;
+	public const FLINT_AND_STEEL = 0x400;
+	public const DIG = self::AXE | self::PICKAXE | self::SHOVEL | self::HOE;
+	public const AXE = 0x800;
+	public const PICKAXE = 0x1000;
+	public const SHOVEL = 0x2000;
+	public const HOE = 0x4000;
+	public const FISHING_ROD = 0x8000;
+	public const SMTH_ON_STICK = 0x10000;
+	public const ELYTRA = 0x20000;
+	public const BRUSH = 0x40000;
+	public const COMPASS = 0x80000;
+	public const RECOVERY_COMPASS = 0x100000;
+	public const HEAD = 0x200000;
```

Comment:
There's a lot of chaos in this code. I don't see why the values have been changed.

The diff seems unnecessarily large.

Reply:
Don't change unnecessary things. It makes it harder to see what's actually been changed.

If your reason for changing something is "it looks better", you shouldn't be doing it alongside code that does something meaningful.

Reply:
Yes, don't add them until they are needed